### ISSUE TITLE
3.11.0: test `GEOSGeomToHEX_buf` fails on s390x

### ISSUE THREAD

**Issue by hjaekel** [libgeos/geos#720]
On s390x (big endian architecture), the test for `GEOSGeomToHEX_buf` fails with geos 3.11.0. Here is the log output of the failure:

```
54/425 Test: unit-capi-GEOSGeomToHEX_buf
Command: "/builds/hjaekel/aports/community/geos/src/geos-3.11.0/bin/test_geos_unit" "capi::GEOSGeomToHEX_buf"
Directory: /builds/hjaekel/aports/community/geos/src/geos-3.11.0/tests/unit
"unit-capi-GEOSGeomToHEX_buf" start time: Nov 01 13:27 UTC
Output:
----------------------------------------------------------
===============================
  GEOS Unit Test Suite
===============================

capi::GEOSGeomToHEX_buf: [1=F]

---> group: capi::GEOSGeomToHEX_buf, test: test<1>
     problem: assertion failed
     failed assertion: `Values are not equal: expected `00000000013FF00000000000004000000000000000` actual `0101000000000000000000F03F0000000000000040``

tests summary: failures:1 ok:0
<end of output>
Test time =   0.02 sec
----------------------------------------------------------
Test Failed.
"unit-capi-GEOSGeomToHEX_buf" end time: Nov 01 13:27 UTC
"unit-capi-GEOSGeomToHEX_buf" time elapsed: 00:00:00
----------------------------------------------------------
```
The same test is successful on 3.10.3. The same test is also successful on all little endian platforms with 3.11.0.

**Comment by hjaekel** [comment]
Please note that expected and actual are reversed in the message above. Therefore I suggest the following change:

```diff
--- a/tests/unit/capi/GEOSGeomToHEX_bufTest.cpp
+++ b/tests/unit/capi/GEOSGeomToHEX_bufTest.cpp
@@ -27,7 +27,7 @@
     std::size_t size{};
     unsigned char* hex = GEOSGeomToHEX_buf(geom1_, &size);
     std::string hexStr(reinterpret_cast<char*>(hex));
-    ensure_equals(std::string{"0101000000000000000000F03F0000000000000040"}, hexStr);
+    ensure_equals(hexStr, std::string{"0101000000000000000000F03F0000000000000040"});
 
     free(hex);
 }
```

**Comment by pramsey** [comment]
So, this commit should "fix" it, but I don't understand why it would ever have worked on an XDR machine, since it was testing a NDR result against an XDR output in all versions. 

2088e3960

**Comment by pramsey** [comment]
Well, the test doesn't even exist in 3.10, so that explains why it doesn't fail. I have back-ported the fix to 3.11 branch, so this is closed now.

**Comment by hjaekel** [comment]
Great, that works. Thank you Paul.
